src/sinks/datadog/metrics/config.rs

```diff
@@ -16,29 +16,35 @@ use crate::{
     http::HttpClient,
     sinks::{
         datadog::DatadogCommonConfig,
-        util::{batch::BatchConfig, ServiceBuilderExt, SinkBatchSettings, TowerRequestConfig},
+        util::{
+            batch::BatchConfig, Concurrency, ServiceBuilderExt, SinkBatchSettings,
+            TowerRequestConfig,
+        },
         Healthcheck, UriParseSnafu, VectorSink,
     },
     tls::{MaybeTlsSettings, TlsEnableableConfig},
 };
 
-// This default is centered around "series" data, which should be the lion's share of what we
-// process.  Given that a single series, when encoded, is in the 150-300 byte range, we can fit a
-// lot of these into a single request, something like 150-200K series.  Simply to be a little more
-// conservative, though, we use 100K here.  This will also get a little more tricky when it comes to
-// distributions and sketches, but we're going to have to implement incremental encoding to handle
-// "we've exceeded our maximum payload size, split this batch" scenarios anyways.
-pub const MAXIMUM_PAYLOAD_COMPRESSED_SIZE: usize = 3_200_000;
-pub const MAXIMUM_PAYLOAD_SIZE: usize = 62_914_560;
-
 // TODO: revisit our concurrency and batching defaults
 const DEFAULT_REQUEST_RETRY_ATTEMPTS: usize = 5;
 
+const DEFAULT_REQUEST_FIXED_CONCURRENCY: usize = 150;
+
 #[derive(Clone, Copy, Debug, Default)]
 pub struct DatadogMetricsDefaultBatchSettings;
 
+// This default is centered around "series" data, which should be the lion's share of what we
+// process. The value of 1k events was arrived at empirically by testing the sink's performance
+// with the v2 series endpoint.
 impl SinkBatchSettings for DatadogMetricsDefaultBatchSettings {
-    const MAX_EVENTS: Option<usize> = Some(100_000);
+    // TODO: the series V1 endpoint support is considered deprecated and should be removed in a future release.
+    // At that time when the V1 support is removed, the 1k value for max events can be used directly.
+    const MAX_EVENTS: Option<usize> = Some(
+        match option_env!("VECTOR_TEMP_USE_DD_METRICS_SERIES_V1_API") {
+            Some(_) => 100_000,
+            None => 1_000,
```

Comment:
I'm concerned about making this 1k given our current aggregation behavior. We only aggregate within a batch and the Datadog backend applies a last-write wins conflict resolution strategy.  Until we implement the final fix (aggregating before batching), changing the batch size from 100k to 1k can result in more write conflicts, which manifests in dropped data.

Reply:
Do you have any suggestions on how I might test that described behavior?
As it stands, leaving with the 100k value resulted in very slow throughput.

Reply:
I don't have a good way to test this. I'm curious if @lukesteensen has thoughts here.

Reply:
What would we be trying to achieve with the test? We know that this is a problem, and the degree to which it manifests is going to be entirely dependent on the input data. You could write a test that shows it very easily, but that would only demonstrate the problem for that test input.

Reply:
My thought with a test was a way to observe the scenario Doug had described, with the thought that the concern could be mitigated by further tuning of the default settings (by limiting the amount of dropped data).

But I didn't realize that

> the degree to which it manifests is going to be entirely dependent on the input data

In which case, I don't understand how we can arrive at a consensus for what the default value should be (which is what my query about a "test", was really getting at - how do we determine what these values should be with enough confidence to release with them.

Reply:
Yeah it's tricky because it's a known bug, which we know we need to fix. In the meantime, we can try to think about the likelihood of having two metric events which should be aggregated be split into different batches. Data loss comes into play when metrics are the same down the the timestamp, so you'd need (for example) two counters to arrive within one second but with more than 500 events (half of 1k) between them to have a >50% chance of seeing data loss in that particular instance. So we can definitely say that reducing the batch size makes the problem more likely to occur at a given throughput level, and significantly lowers the throughput level at which it's likely users could see the issue.

I'm curious if we have dug into why exactly a large batch size causes such a slowdown. That seems like something that's important to understand, since it's relatively likely we'll have users try to increase that number.

Reply:
Thanks for that additional explanation 🙇 

> I'm curious if we have dug into why exactly a large batch size causes such a slowdown. That seems like something that's important to understand, since it's relatively likely we'll have users try to increase that number.

I haven't dug into that yet no. I was just modifying a fixed concurrency and batch max_events (with a large batch timeout), to see if I could maximize throughput. This gets back to the observation from last week that we really need _something_ to aid in making it more clear how all of this interacts and what considerations should be taken.

Reply:
:hmmm: so I'm running the test again but with the 100_000 max_events , and adaptive concurrency, and I'm seeing better throughput than what I was seeing with the fixed concurrency experiment 😭